packmanager/umbrella.go

```diff
@@ -123,6 +126,55 @@ func (u umbrella) AddSource(ctx context.Context, source string) error {
 	return nil
 }
 
+func (u umbrella) Prune(ctx context.Context, qopts ...QueryOption) error {
+	var query Query
+	var tarFiles []fs.DirEntry
+	var errList []error
+	homeDir, err := os.UserHomeDir()
+	sourcesDir := homeDir + "/.local/share/kraftkit/sources/"
+	for _, qopt := range qopts {
+		qopt(&query)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if query.all {
+		if err != nil {
+			return err
+		}
+		tarFiles, err = os.ReadDir(sourcesDir)
+		if err != nil {
+			return err
+		}
+
+		for _, tarFile := range tarFiles {
+			if tarFile.Name() == ".unikraft" {
+				continue
+			}
+			err := os.Remove(sourcesDir + tarFile.Name())
+			if err != nil {
+				errList = append(errList, err)
+			}
+		}
+
+		if len(errList) > 0 {
+			err = errors.Join(errList...)
+			return err
+		}
+	} else {
+		packPath := sourcesDir + query.name + "-" + query.version + ".tar.gz"
```

Comment:
The source is ultimately handled by the [`ManifestProvider`](https://github.com/unikraft/kraftkit/blob/staging/manifest/provider.go#L17-L27). A new method `DeleteManifest` should be added and called following a successful lookup. This allows for different manifest implementations to correctly handle the removal of the artifact.

Reply:
Done 👍🏽